python_modules/libraries/dagster-dask/dagster_dask/data_frame.py

```diff
@@ -29,852 +33,308 @@
         },
-    )
-)
-def dataframe_loader(_context, config):
-    file_type, file_options = list(config.items())[0]
-    path = file_options.get("path")
+    })
+
 
-    if file_type == "csv":
-        return dd.read_csv(path, **dict_without_keys(file_options, "path"))
-    elif file_type == "parquet":
-        return dd.read_parquet(path, **dict_without_keys(file_options, "path"))
-    elif file_type == "hdf":
-        return dd.read_hdf(path, **dict_without_keys(file_options, "path"))
-    elif file_type == "json":
-        return dd.read_json(path, **dict_without_keys(file_options, "path"))
-    elif file_type == "sql_table":
-        return dd.read_sql_table(**file_options)
-    elif file_type == "table":
-        return dd.read_table(path, **dict_without_keys(file_options, "path"))
-    elif file_type == "fwf":
-        return dd.read_fwf(path, **dict_without_keys(file_options, "path"))
-    elif file_type == "orc":
-        return dd.read_orc(path, **dict_without_keys(file_options, "path"))
+@dagster_type_loader(_dataframe_loader_config())
+def dataframe_loader(_context, config):
+    read_type, read_options = None, None
+    if "read" in config:
+        read_type, read_options = next(iter(config["read"].items()))
     else:
+        for key in DataFrameReadTypes:
+            if key in config:
+                read_type, read_options = key, config[key]
+                warnings.warn("Specifying {key}: is deprecated. Use read:{key}: instead.".format(key=key))
+    if not read_type:
+        raise DagsterInvariantViolationError(
+            "No read_type found. Expected read key in config."
+        )
+    if not read_type in DataFrameReadTypes:
         raise DagsterInvariantViolationError(
-            "Unsupported file_type {file_type}".format(file_type=file_type)
+            "Unsupported read_type {read_type}.".format(read_type=read_type)
         )
 
+    # Get the metadata entry for the read_type in order to know which function
+    # to call and whether it uses path as the first argument. And, make
+    # read_options mutable if we need to pop off a path argument.
+    read_meta = DataFrameReadTypes[read_type]
+    read_options = dict(read_options)
+    
+    # Get the read function and prepare its arguments.
+    read_function = read_meta["function"]
+    read_args = [read_options.pop("path")] if read_meta.get("is_path_based", False) else []
+    read_kwargs = read_options
+    
+    df = read_function(*read_args, **read_kwargs)
+
+    return df
+
+
+def _dataframe_materializer_config():
+    to_fields = {
+        write_to: Permissive({
+            option_name: Field(option_args[0], is_required=option_args[1], description=option_args[2])
+            for option_name, option_args in to_opts["options"].items()
+        })
+        for write_to, to_opts in DataFrameToTypes.items()
+    }
+
+    return Shape({
+        "to": Field(
+            Selector(to_fields),
+            is_required=False,
+        ),
+        **{
```

Comment:
same as above - i think this should be `Selector`

Reply:
Changed to `Selector` for this PR.